pkg/cloudprovider/suite_test.go

```diff
@@ -314,6 +327,23 @@ var _ = Describe("CloudProvider", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(isDrifted).To(BeTrue())
 		})
+		It("should return drifted if the securitygroup is not valid", func() {
```

Comment:
Can we make sure we cover all the edge cases? What about when the security groups on the instance are a subset of the valid security groups?